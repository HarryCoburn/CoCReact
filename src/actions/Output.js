import * as OutputMsg from "./messages/outputMsg";

/**
 * Message sender for updating the text in the view.
 * @param {JSX} payload
 * @return {object} Redux action
 */
export function _updateView(payload) {
  if (payload === undefined || payload === null) {
    throw Error("Core._updateView recevied undefined or null text ouput");
  }
  return {
    type: OutputMsg.CHANGE_TEXT,
    payload
  };
}

export function _addText(payload) {
  if (payload === undefined || payload === null) {
    throw Error("Core._addText recevied undefined or null text ouput");
  }
  return {
    type: OutputMsg.ADD_TEXT,
    payload
  };
}
