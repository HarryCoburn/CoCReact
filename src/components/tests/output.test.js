import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Output from "../output";

const mockStore = configureStore([]);

describe("Output Component", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      output: <p>This is some output</p>
    });

    component = renderer.create(
      <Provider store={store}>
        <Output />
      </Provider>
    );
  });

  xit("should match the snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
