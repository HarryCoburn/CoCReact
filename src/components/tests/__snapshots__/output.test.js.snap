// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output Component should match the snapshot 1`] = `
<div
  id="maintext"
/>
`;
